src/components/button/Button.js

```diff
@@ -0,0 +1,44 @@
+import './styles.css';
+
+class Button {
```

Comment:
I tried to change it that way, but I'm getting an error about "illegal constructor".
![image](https://user-images.githubusercontent.com/80357638/146460398-ed0e1e9e-4ba0-4c3e-a340-01319292a03a.png)

I found on stackoverflow that it can be related to `HTMLElement` is an abstract class.
https://stackoverflow.com/questions/41521812/illegal-constructor-with-ecmascript-6

Reply:
Have you tried something like this?
```
import HtmlElement from '../htmlElement/HtmlElement';
import './styles.css';

class Button extends HtmlElement {
  constructor({ label = '', onClick = null, buttonType = 'default', imageSrc = null, id, width = 'auto' }) {
    super(['button'], 'button', label);
    this.id = id;
    this.onClick = onClick;
    this.buttonType = buttonType;
    this.imageSrc = imageSrc;
    this.width = width;
  }

  render() {
    const btn = super.render();
    if (this.imageSrc) {
      const img = document.createElement(`img`);
      img.src = this.imageSrc;
      btn.appendChild(img);
    }

    if (this.buttonType === 'default') {
      btn.classList.add('default');
    }
    if (this.buttonType === 'hof') {
      btn.classList.add('hof');
    }
    if (this.buttonType === 'ptg') {
      btn.classList.add('ptg');
    }
    // could be shortened to:
    // btn.classList.add(this.buttonType);

    if (this.width === 'auto') {
      btn.style.width = 'auto';
    }
    if (this.width === 'full') {
      btn.style.width = '100%';
    }
    if (this.width === 'half') {
      btn.style.width = '50%';
    }
    btn.addEventListener('click', this.onClick);

    return btn;
  }
}

export default Button;
```
You don't have to go that far with the changes if you decide to apply them. It's an attempt to make the most of inheriting from `HtmlElement`.